### DISCUSSION TITLE
Explicit sorting of prevalues in property editors

### DISCUSSION THREAD

**Discussion by bjarnef** [umbraco/Umbraco-CMS#13065] [Features and ideas]
When defining prevalues for a property editor using `ConfigurationEditor` and the `ConfigurationField` attribute it seems the prevalues are ordered by the index, but simple enough to re-order the fields in the class.

```
internal class ButtonPickerConfigurationEditor : ConfigurationEditor<ButtonPickerConfiguration>
{

}
```

```
public class ButtonPickerConfiguration
{
    [ConfigurationField("defaultValue", "Default value", "textstring", Description = "If no default value selected, first value is used.")]
    public string? DefaultValue { get; set; }

    [ConfigurationField("buttons", "Buttons", "multivalues", Description = "The defined buttons.")]
    public List<ValueListItem> Buttons { get; set; } = new();

    [ConfigurationField("size", "Size", "textstring", Description = "Size of the buttons.")]
    public string? Size { get; set; }
}
```

However when inheriting from another configuration, e.g. `ValueListConfiguration`, which for instance the `DropDownFlexibleConfiguration` inherits https://github.com/umbraco/Umbraco-CMS/blob/v10/contrib/src/Umbraco.Core/PropertyEditors/DropDownFlexibleConfiguration.cs#L3 it seems there isn't a property to explicit set the sort order like we are familiar with in Content Apps and Dashboards.

```
public class ButtonPickerConfiguration : ValueListConfiguration
{
    [ConfigurationField("defaultValue", "Default value", "textstring", Description = "If no default value selected, first value is used.")]
    public string? DefaultValue { get; set; }

    [ConfigurationField("size", "Size", "textstring", Description = "Size of the buttons.")]
    public string? Size { get; set; }
}
```

Sometimes developers want to group the configuration fields, so it may not always make sense to place inherited "Items" first or last.

I found a way to sort the Fields alfabetically by `Key` though:

```
public ButtonPickerConfigurationEditor(
        IIOHelper ioHelper,
        IEditorConfigurationParser editorConfigurationParser)
        : base(ioHelper, editorConfigurationParser)
    {
        ConfigurationField items = Fields.First(x => x.Key == "items");
        items.Name = "Buttons";
        items.Description = "The defined buttons.";

        Fields.Sort((x, y) => string.Compare(x.Key, y.Key));

        // customize the items field
        items.Validators.Add(new ValueListUniqueValueValidator());
    }
```

However I would be great is we explicit could set the sort order or weight instead maybe similar to sort order on `[Tree]` attribute:
https://github.com/umbraco/Umbraco-CMS/blob/v10/contrib/src/Umbraco.Web.BackOffice/Trees/TreeAttribute.cs#L45-L48

```
[ConfigurationField("buttons", "Buttons", "multivalues", Description = "The defined buttons.", SortOrder = 10)]
public List<ValueListItem> Buttons { get; set; } = new();
```

**Comment by nul800sebastiaan** [comment]
Fixed in https://github.com/umbraco/Umbraco-CMS/pull/13075
